pkg/api/controller_test.go

```diff
@@ -5214,7 +5214,7 @@ func TestManifestImageIndex(t *testing.T) {
 				Layers:   layers,
 				Manifest: manifest,
 			}, baseURL, repoName)
-		So(err, ShouldNotBeNil)
+		So(err, ShouldBeNil)
```

Comment:
Why did these tests need to change?

Reply:
I have changed the way UploadImage works, now if no reference is provided it is assumed that the digest of the manifest will be the reference. The test I changed was checking ShouldNotBeNil just so that it passes and I think who wrote it didn't have time to investigate what's the problem but semantically I see that the test just wants to push images

Reply:
Can you double check? There should be some tests checking that for a given tag you cannot overwrite an image index with an image manifest, and the other way around. Not sure if these are those tests of those are in another place.

Reply:
Yes, the test you're reffering to is called `Convey("Change media-type", func() {...}` and it wasn't changed.

Reply:
Ok, thanks